code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -140,7 +140,19 @@
 		throwpower = I.throwforce
 		if(I.thrownby == src) //No throwing stuff at yourself to trigger hit reactions
 			return ..()
-	if(check_shields(throwpower, "\the [AM.name]", AM, THROWN_PROJECTILE_ATTACK))
+	if(martial_art && martial_art.catch_chance)

```

Comment:
move all of this to a proc on the martial arts datums and change it to

    if(martial_art && martial_art.catch_item)
        return TRUE